### DISCUSSION TITLE
How to disable %preuntrans & %sysusers creation while building an rpm

### DISCUSSION THREAD

**Discussion by sshedi** [rpm-software-management/rpm#3174] [General]
I'm yet to root cause this issue because this started happening today and I have no idea how or why.

```
D: running %preuntrans scripts
D: ========== +++ systemd-255.2-5.ph5 x86_64-linux 0x2
D:   install: systemd-255.2-5.ph5.x86_64 has 732 files
D: %sysusers(systemd-255.2-5.ph5.x86_64): scriptlet start
D: %sysusers(systemd-255.2-5.ph5.x86_64): execv(/bin/sh) pid 3927
+ /usr/lib/rpm/sysusers.sh --replace /usr/lib/sysusers.d/basic.conf -
/var/tmp/rpm-tmp.JhzjTt: line 2: /usr/lib/rpm/sysusers.sh: No such file or directory
D: %sysusers(systemd-255.2-5.ph5.x86_64): waitpid(3927) rc 3927 status 7f00
error: %sysusers(systemd-255.2-5.ph5.x86_64) scriptlet failed, exit status 127
package systemd-255.2-5.ph5.x86_64: script error in (unknown)
error: systemd-255.2-5.ph5.x86_64: install failed
D: ========== +++ systemd-255.2-4.ph5 x86_64-linux 0x2
```

I'm not adding anything like this in my systemd.spec but somehow while installing rpm is looking for sysusers.sh.
I have sysusers.sh packaged in my rpm.spec but what I don't understand is how it started showing up all of a sudden.

Link to my systemd spec
https://github.com/vmware/photon/blob/master/SPECS/systemd/systemd.spec

Link to sysusers.sh
https://github.com/vmware/photon/blob/master/SPECS/rpm/rpm.spec#L262

I'm using `rpm-4.19.1.1`
I commented out `__systemd_sysusers` from /lib/rpm/macros and then update goes smooth as expected.

But is there anyway to avoid embedding this info in rpm (systemd rpm in this case) during build time itself?

Please assist.

**Comment by sshedi** [comment]
Decided to comment out `__systemd_sysusers` from macros file.
https://github.com/vmware/photon/blob/dev/SPECS/rpm/0006-disable-sysusers.patch#L26

I didn't find a way to exclude this info from getting embed in the built rpm.

**Comment by pmatilai** [comment]
This is a headline feature of >= 4.19: https://rpm-software-management.github.io/rpm/manual/users_and_groups.html 

The clue is right there:
> /var/tmp/rpm-tmp.JhzjTt: line 2: /usr/lib/rpm/sysusers.sh: No such file or directory

You're not packaging the sysusers.sh script which comes with rpm 4.19, which is what that %__systemd_sysusers points to by default. Yes, commenting out the macro will disable the feature if that's what you want. The user/group provides are generated like any other automatic dependency: https://rpm-software-management.github.io/rpm/manual/dependency_generators.html

**Reply by sshedi** [reply]
Thanks for the response and details. Yes, I managed to figure it out after a while.
The issue is, there are few systems which have `rpm-libs` package but not `rpm` so, that is where this issue started showing up.
The system has `rpm-libs` and a package manager `tdnf`. Probably I should package sysusers.sh with `rpm-libs` instead of rpm.

**Reply by pmatilai** [reply]
That means rpm is incorrectly packaged, yeah. The rpm libraries need much more than just themselves for normal operation - rpmrc, main macros, platform macros.. and now the sysusers.sh script (or systemd-sysusers if you configure it to use that). In Fedora/RHEL rpm-libs simply requires rpm where all the config stuff is, but if size is critical then I'd split all that to rpm-common subpackage. Of course if multilib is never a concern then all that goo can go to the library package too.
